engine/src/main/java/org/pentaho/di/shared/RepositorySharedObjectsIO.java

```diff
@@ -0,0 +1,239 @@
+/*!
+ * Copyright 2024 Hitachi Vantara.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.pentaho.di.shared;
+
+import org.pentaho.di.cluster.ClusterSchema;
+import org.pentaho.di.cluster.ClusterSchemaManagementInterface.SlaveServersSupplier;
+import org.pentaho.di.cluster.SlaveServer;
+import org.pentaho.di.core.Const;
+import org.pentaho.di.core.database.DatabaseMeta;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.partition.PartitionSchema;
+import org.pentaho.di.repository.ObjectId;
+import org.pentaho.di.repository.Repository;
+import org.pentaho.di.repository.RepositoryElementInterface;
+import org.pentaho.di.repository.RepositoryExtended;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.w3c.dom.Node;
+
+/**
+ * An implementation of SharedObjectsIO that backs to a Repository.
+ * <p>
+ * This class does not cache anything. Note that PurRepository does its own caching, but only through the
+ * RepositoryExtended interface, which this class makes use of.
+ *
+ */
+public class RepositorySharedObjectsIO implements SharedObjectsIO {
+
+  private final Repository repository;
+  private final SlaveServersSupplier slaveServerSupplier;
+
+  public RepositorySharedObjectsIO( Repository repository, SlaveServersSupplier slaveServerSupplier ) {
+    this.repository = Objects.requireNonNull( repository );
+    this.slaveServerSupplier = slaveServerSupplier;
+  }
+
+  @Override
+  public Map<String, Node> getSharedObjects( String type ) throws KettleException {
+    SharedObjectType objectType = SharedObjectType.valueOf( type.toUpperCase() );
+    List<? extends SharedObjectInterface> objects = null;
+    if ( repository instanceof RepositoryExtended ) {
```

Comment:
In what case do you get the repositry that is instance of RepositoryExtended

Reply:
FileRepository does not implement RepositoryExtended, but PurRepository does.